spring-pulsar-spring-cloud-stream-binder/src/main/java/org/springframework/pulsar/spring/cloud/stream/binder/provisioning/PulsarTopicProvisioner.java

```diff
@@ -34,74 +36,63 @@
 public class PulsarTopicProvisioner implements
 		ProvisioningProvider<ExtendedConsumerProperties<PulsarConsumerProperties>, ExtendedProducerProperties<PulsarProducerProperties>> {
 
-	// TODO: Retrieve partitions through config
+	private final PulsarAdministration pulsarAdministration;
+
+	private final PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties;
+
+	public PulsarTopicProvisioner(PulsarAdministration pulsarAdministration,
+			PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties) {
+		this.pulsarAdministration = pulsarAdministration;
+		this.pulsarBinderConfigurationProperties = pulsarBinderConfigurationProperties;
+	}
+
 	@Override
 	public ProducerDestination provisionProducerDestination(String name,
 			ExtendedProducerProperties<PulsarProducerProperties> properties) throws ProvisioningException {
-		PulsarTopic pulsarTopic = PulsarTopic.builder(name).numberOfPartitions(1).build();
-		return new PulsarProducerDestination(pulsarTopic.topicName(), pulsarTopic.numberOfPartitions());
+
+		int partitionCount = this.pulsarBinderConfigurationProperties.partitionCount();
+		int partitionCountOnBinding = properties.getPartitionCount(); // leverage the
+																		// partition count
+																		// from core
+		if (partitionCountOnBinding > 1) {
+			partitionCount = partitionCountOnBinding;
+		}
+		PulsarTopic pulsarTopic = PulsarTopic.builder(name).numberOfPartitions(partitionCount).build();
+		this.pulsarAdministration.createOrModifyTopics(pulsarTopic);
+		return new PulsarDestination(pulsarTopic.topicName(), pulsarTopic.numberOfPartitions());
 	}
 
 	@Override
 	public ConsumerDestination provisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<PulsarConsumerProperties> properties) throws ProvisioningException {
-		PulsarTopic pulsarTopic = PulsarTopic.builder(name).numberOfPartitions(1).build();
-		return new PulsarConsumerDestination(pulsarTopic.topicName(), pulsarTopic.numberOfPartitions());
+		int partitionCount = this.pulsarBinderConfigurationProperties.partitionCount();
+
+		int partitionCountOnBinding = properties.getExtension().getPartitionCount(); // partition
+																						// count
```

Comment:
**[NIT]** The formatter does this on the end of line comments. Do you mind moving it up on the line above the code? 

One of the most confusing areas of SCSt (for me anyways) is the extended props. I would love to see a comment like this: 

```
// The spring.cloud.stream.pulsar.bindings.<NAME>.consumer.partition-count 
```